File: go-backend/models/file.go
package models

import "time"

type File struct {
	ID        int         `json:"id"`
	UserID    int         `json:"user_id"`
	Name      string      `json:"name"`
	Filetype  string      `json:"filetype"`
	Path      string      `json:"path"`
	Filename  string      `json:"filename"`
	Size      int         `json:"size"`
	CreatedBy int         `json:"created_by"`
	UpdatedBy int         `json:"updated_by"`
	CardPK    int         `json:"card_pk"`
	IsDeleted bool        `json:"is_deleted"`
	CreatedAt time.Time   `json:"created_at"`
	UpdatedAt time.Time   `json:"updated_at"`
	Card      PartialCard `json:"card"`
}

type EditFileMetadataParams struct {
	Name   string `json:"name"`
	CardPK int    `json:"card_pk"`
}

type UploadFileResponse struct {
	Message string `json:"message"`
	File    File   `json:"file"`
}

type UploadFileParams struct {
	Filename    string `json:"filename"`
	ContentType string `json:"content_type"`
}

File: zettelkasten-front/src/models/File.ts
// src/models/File.ts
import { PartialCard } from "./Card";

export interface File {
  id: number;
  name: string;
  filetype: string;
  path: string;
  filename: string;
  size: number;
  created_by: number;
  updated_by: number;
  card_pk: number;
  is_deleted: boolean;
  created_at: string;
  updated_at: string;
  card: PartialCard;
}

export interface UploadFileResponse {
  message: string;
  file: File;
}

export interface EditFileMetadataParams {
  name: string;
}